expression/builtin_json_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/util/testleak"
+	"github.com/pingcap/tidb/util/testutil"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tidb/util/types/json"
+)
+
+func (s *testEvaluatorSuite) TestJSONType(c *C) {
+	defer testleak.AfterTest(c)()
+	fc := funcs[ast.JSONType]
+	tbl := []struct {
+		Input    interface{}
+		Expected interface{}
+	}{
+		{nil, nil},
+		{`3`, `INTEGER`},
+		{`3.0`, `DOUBLE`},
+		{`null`, `NULL`},
+		{`true`, `BOOLEAN`},
+		{`[]`, `ARRAY`},
+		{`{}`, `OBJECT`},
```

Comment:
those type cannot be used here. If you run `select json_type(datetime_field)` in mysql 5.7,
error will be return.

`json_type` can only retreive `json` or `string` as argument.